frontend/src/components/ModalSetUsername/ModalSetUsername.tsx

```diff
@@ -0,0 +1,50 @@
+import { Button, Dialog, DialogActions, DialogContent, TextField, Typography } from '@material-ui/core'
+import React from 'react'
+import useApplication from '../../shared/applications'
+
+export const ModalSetUsername: React.FC<{
+  connection?: IConnection
+  openUsername: boolean
+  handleSubmit: () => void
+  username: string
+  setUsername: React.Dispatch<React.SetStateAction<string>>
+  service?: IService
+  close: () => void
```

Comment:
```suggestion
  connection?: IConnection
  open: boolean
  service?: IService
  onSubmit: (username: string) => void
  onClose: () => void
```
Let's have the modal keep the username state and return the username when it's submitted.
I don't think we need to say username again on the open because we know it's the username modal.
And I like to follow the material UI convention of prefixing any callbacks with "on" so it becomes onClose and onSubmit.